packages/eslint-plugin/src/rules/no-state-access.ts

```diff
@@ -88,21 +72,42 @@ const rule: TSRuleModule = {
       Identifier(node: TSESTree.Identifier): void {
         if (
           !importsStateFromModel ||
-          insideFunction.length !== 0 ||
+          insideFunctionStackDepth !== 0 ||
           insideImportDeclaration
         ) {
           return;
         }
-        const identifierTsNode = parserServices.esTreeNodeToTSNodeMap!.get<
-          TSNode
-        >(node);
-        const symbol = checker.getSymbolAtLocation(identifierTsNode);
-        if (symbol && symbol === stateImportSymbol) {
+        if (
+          stateImportNode &&
```

Comment:
It would be nice to try and generalise this rule to everything in the view and action contexts. Happy for this to happen in another PR, but might be worth thinking about it here.

Reply:
See the babel-plugin for how you might be able to adapt the logic.